code/modules/status_system/statusEffects.dm

```diff
@@ -2200,3 +2200,34 @@
 	onRemove()
 		. = ..()
 		owner.remove_filter("gnesis_tint")
+
+/datum/statusEffect/portal_weakness
+	id = "portal_weakness"
+	name = "Dark Influence"
+	desc = "An ominous aura instills fear in you. You feel your grasp on life slipping. Maximum Health and Stamina lowered."
+	icon_state = "maxhealth-"
+	unique = 1
+	maxDuration = 1 MINUTES
+	var/max_health = 30
+	var/max_stam = -30
+	var/mob/living/carbon/human/H
+
+	onAdd(optional=null)
+		. = ..()
+		if (!ishuman(owner))
+			owner.delStatus("portal_weakness")
+		else
+			H = owner
+			H.max_health -= max_health
+			health_update_queue |= H
+			H.add_stam_mod_max("portal_weakness", max_stam)
+
+	onRemove()
+		. = ..()
+		if (ishuman(owner))
```

Comment:
Should probably `H = null` after this. Or just don't have H as a member variable but as a local on both procs.